### PR TITLE
Add code coverage report generation for debugger integration tests

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR integrates jacoco based codecov report generation for debugger integration tests.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/31352.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (praveennadarajah) [inline tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java:2]:
```suggestion
 *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    public OutputStream getOutputStream() {
+        return outputStream;
+    }
+
+    // Todo - Implement
+
+    //    @Override
+    //    public InputStream getErrorStream() {
+    //        inputStream;
+    //    }
```
Comment (praveennadarajah) [inline tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java:148]: Is this intentional?

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    @Override
+    public OutputStream getOutputStream() {
+        return outputStream;
+    }
+
+    // Todo - Implement
+
+    //    @Override
+    //    public InputStream getErrorStream() {
+    //        inputStream;
```
Comment (praveennadarajah) [inline tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java:147]:
```suggestion
    //        return inputStream;
```

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java:2]: This class just got renamed so we can keep the original year IMO

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java (hunk 2 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java]: Removed

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java (hunk 3 above)
Comment (NipunaRanasinghe) [reply tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/client/connection/SocketStreamConnectionProvider.java]: Removed the commented lines

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**